### PR TITLE
[Projects] Support Detic inference.

### PR DESCRIPTION
**Pull request by RangiLyu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Support Detic inference

image

## TODO

- [x] Align inference accuracy.
- [x] Support custom vocabulary demo.

### CODE REVIEW

**Review by ZwwWayne — DISMISSED** [review]

**Review by ZwwWayne — COMMENTED** [review]
Overall, this PR LGTM

File: projects/Detic/README.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+## Results
+
+Here we provide the baseline version of SparseInst with ResNet50 backbone.
+
+To find more variants, please visit the [official model zoo](https://github.com/facebookresearch/Detic/blob/main/docs/MODEL_ZOO.md).
+
+| Backbone |    Training data    | Lr schd | Mem (GB) | FPS | mask AP |                     Config                      |        Download        |
+| :------: | :-----------------: | :-----: | :------: | :-: | :-----: | :---------------------------------------------: | :--------------------: |
+|   R-50   | ImageNet-21K & LVIS |    -    |    -     |  -  |  32.4   | [config](./configs/detic_centernet2_r50_fpn.py) | [model](-) \| [log](-) |
```
Comment (ZwwWayne) [inline projects/Detic/README.md:81]: do we need to convert some model or provide model convert script？

File: projects/Detic/detic/utils.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        text_encoder.eval()
+        texts = [prompt_prefix + x for x in custom_vocabulary]
+        embeddings = text_encoder(texts).detach().permute(
+            1, 0).contiguous().cpu()
+        return embeddings
+    else:
+        raise NotImplementedError
+
+
+DATASET_EMBEDDINGS = {
```
Comment (ZwwWayne) [inline projects/Detic/detic/utils.py:28]: also add comments to.indicate where to get them.

**Review by ZwwWayne — APPROVED** [review]
